app/scripts/models/reliers/oauth.js

```diff
@@ -157,6 +163,21 @@ define(function (require, exports, module) {
       }
     },
 
+    _importValidateOptionalSearchParam: function (sourceName, destName, validateFunc){
+      var self = this;
+      var sourceValue = self.getSearchParam(sourceName);
+      // Only validate if param is in window.location
+      if (sourceValue){
+        if (validateFunc(sourceValue)){
+          self.importSearchParam(sourceName, destName);
+        } else {
+          var err = OAuthErrors.toError('INVALID_PARAMETER');
```

Comment:
Other than the OAuth specific error code, this function makes sense on the base relier so it can be used by others.